### PR TITLE
Deprecating global $pmpro_levels

### PR DESCRIPTION
**Pull request by dparker1005**
### All Submissions:

* [x] Have you followed the [Contributing guideline](https://github.com/strangerstudios/paid-memberships-pro/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/strangerstudios/paid-memberships-pro/pulls) for the same update/change?

### Changes proposed in this Pull Request:
Removing uses of the `$pmpro_levels` global besides for those needed for backwards compatibility. The `pmpro_getAllLevels()` function should be used instead.

Deprecating that global also allows simplifying the arguments for the `pmpro_getAllLevels()` function as the second `$use_cache` argument was unclear.

Looking at our Add Ons, it does not any use the `$pmpro_levels` global without setting it using `pmpro_getAllLevels()`, so backwards compatibility with our Add Ons should not be an issue. The following regex can be used to find all instances of the global being defined: `^(?=.*\bglobal\b)(?=.*\$pmpro_levels).*`

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: preheaders/checkout.php [hunk 1]
```diff
@@ -819,6 +815,9 @@
 	}
 }
 
+// Preventing conflicts with old checkout templates that depend on the $pmpro_level global being set.
+pmpro_getAllLevels();
```
Comment (kimcoleman) [inline preheaders/checkout.php:819]: Curious how to test just this one piece. Why we need this and how we'd know it is or isn't needed.

File: shortcodes/pmpro_account.php [hunk 2]
```diff
@@ -4,7 +4,7 @@
 */
 function pmpro_shortcode_account($atts, $content=null, $code="")
 {
-	global $wpdb, $pmpro_msg, $pmpro_msgt, $pmpro_levels, $current_user, $levels;
+	global $wpdb, $pmpro_msg, $pmpro_msgt, $current_user, $levels;
```
Comment (kimcoleman) [inline shortcodes/pmpro_account.php:7]: I don't see $levels used anywhere in this. Perhaps that global isn't needed anymore either? Just asking - could be a separate PR.

**Review by kimcoleman — CHANGES_REQUESTED** [review]
I have a couple comments in here to chat through but mostly otherwise approved.

File: preheaders/checkout.php (hunk 1 above)
Comment (kimcoleman) [reply preheaders/checkout.php:819]: Confirming this is needed and resolving.

File: shortcodes/pmpro_account.php (hunk 2 above)
Comment (kimcoleman) [reply shortcodes/pmpro_account.php:7]: Going to remove $levels but also I don't think we need pmpro_msg or pmpro_msgt. Removing both - we will see if we find that these are needed. This code wasn't touched since 8+ years ago. I think it isn't used.

**Comment by Leinad4Mind** [comment]
Just one doubt, is global $pmpro_level deprecated too? Or only the plural one?

**Comment by dparker1005** [comment]
The `$pmpro_level` global is still available at checkout, but we recommend switching to using `$pmpro_level = pmpro_getLevelAtCheckout();` to make sure that the level object has the correct unaltered information.
